src/Projection/InMemoryEventStoreQueryFactory.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * This file is part of the prooph/event-store.
+ * (c) 2014-2017 prooph software GmbH <contact@prooph.de>
+ * (c) 2015-2017 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Projection;
+
+use Prooph\EventStore\EventStore;
+
+final class InMemoryEventStoreQueryFactory implements QueryFactory
+{
+    /**
+     * @var EventStore
+     */
+    private $eventStore;
+
+    public function __construct(EventStore $eventStore)
+    {
+        $this->eventStore = $eventStore;
+    }
+
```

Comment:
@prolic The event store is mandatory in constructor. Do I miss something?

```php
    public function __construct(EventStore $eventStore)
    {
        $this->eventStore = $eventStore;
    }
```

A better solution is this, right? No constructor, no setter, no cry. :-) This would be a clean interface, same for the other factories.

```php
interface QueryFactory
{
    public function __invoke(EventStore $eventStore, ProjectionOptions $options = null): Query;
}
```